src/manage-apps/fix-packages.md

```diff
@@ -0,0 +1,68 @@
+# Cleaning Up Packages with Terminal Commands
+
+If your system complains about a failed upgrade, package manager conflicts, broken upgrades, or other package-related issues, there are several common fixes to these problems. Some package manager issues can be resolved with the graphical update program, but many require the command line.
+
+## General Fixes for Apt Packages
+
+If a package was installed with `apt`, these commands perform general cleanup that can resolve many of `apt`'s errors. These commands should be run one at a time.
+
+```bash
+sudo apt clean
+sudo apt update -m
+sudo dpkg --configure -a
+sudo apt install -f
+sudo apt full-upgrade
+sudo apt autoremove --purge
+```
+
+- `apt clean` - The `clean` command clears out the local repository of retrieved package files.
+- `apt update -m` - The `-m` option ignores corrupted package files during a package update.
```

Comment:
I held my tongue on this on the last read-through, but I think we might want to examine what `-m` is doing here a little closer. `apt update` doesn't fetch package files (it only fetches indexes), so ignoring corrupted package files doesn't make sense. But `update` doesn't accept all of the flags that `upgrade`/`full-upgrade`/`install` do, so it must be doing something. My understanding was that when used with `update`, the `-m` is just there to not give a failed exit code if any of the configured repositories were missing/invalid. (The user still sees the error message, but a script would continue instead of stopping.) I'll need to verify this.

Reply:
I found this in the apt-get manpage (nothing of the -m opton is mentioned in the `apt` manpage): 

`  -m, --ignore-missing, --fix-missing
           Ignore missing packages; if packages cannot be retrieved or fail
           the integrity check after retrieval (corrupted package files), hold
           back those packages and handle the result. Use of this option
           together with -f may produce an error in some situations. If a
           package is selected for installation (particularly if it is
           mentioned on the command line) and it could not be downloaded then
           it will be silently held back.
`

Reply:
That description's not accurate either, since apt will still fetch the rest of the indexes if one of the configured repos is invalid, even without this option.

It's not the return value like I thought. If I configure a valid domain with an invalid suite (leading to a 404 error), I get a 100 return code whether I'm using the option or not. (And if I use an invalid domain leading to a "could not resolve" error, I get a 0 return code either way.)

Reply:
I found a relevant bug report. It seems that `-m` used to keep the return code at 0 (success) in the case of fetching errors like I'd heard, but that stopped working a very long time ago: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=260146 

From a Debian developer in that bug thread:

> I do not think -m/--fix-missing has ever been applicable to apt-get update. As documented, it only applies to the downloading of packages (not package indexes).

I believe I've confirmed this in apt's code. I see a few places in the `apt install` code where `APT::Get::Fix-Missing` is checked: https://salsa.debian.org/apt-team/apt/-/blob/6499f7dc842ebd621f11ea01e83ccbe4125a0581/apt-private/private-install.cc 

But I don't see it checked anywhere in the `apt update` code: https://salsa.debian.org/apt-team/apt/-/blob/6499f7dc842ebd621f11ea01e83ccbe4125a0581/apt-private/private-update.cc 

My remaining question is why `apt update` doesn't error with `-m`. In the [code where it's taken as a command line argument](https://salsa.debian.org/apt-team/apt/-/blob/6499f7dc842ebd621f11ea01e83ccbe4125a0581/apt-private/private-cmndline.cc#L267), there's a `FIXME: move to the correct command(s)` comment above the block. Other commands in this block that seem irrelevant are also taken by `apt update` without error.

@Ampersandstorm With this in mind, I think it's safe to assume we can just drop `-m` from the suggested command (and description). I think it was added to the support article by mistake or misunderstanding a long time ago, and we've just assumed it had purpose ever since. (I checked the git history; this command was added along with four of the others in the list, and the person who added them was gone before I got here.)

Reply:
This has been updated, resolving unless you discover additional issues.